pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/memory/PinotDataBuffer.java

```diff
@@ -152,17 +153,22 @@ public static PinotBufferFactory getFactory() {
   }
 
   public static PinotBufferFactory createDefaultFactory() {
-    return createDefaultFactory(true);
+    String skipBbEnvValue = System.getenv(SKIP_BYTEBUFFER_ENV);
+    boolean prioritizeByteBuffer = !Boolean.parseBoolean(skipBbEnvValue);
```

Comment:
Can we just parse this once? 
Make `prioritizeByteBuffer` a static value is better?

Reply:
Changed.

But remember this is called only once per process in production code, so there is no actual gain.